src/core/receive.rs

```diff
@@ -104,8 +106,10 @@ impl ReceiveMachine {
         phase: &Phase,
         body: &[u8],
     ) -> Option<Vec<u8>> {
-        let data_key = key::derive_phase_key(&side, &key, &phase);
-        key::decrypt_data(&data_key, body)
+        let mut data_key = key::derive_phase_key(&side, &key, &phase);
```

Comment:
You can use `Zeroizing` to do zeroize-on-drop automatically: https://docs.rs/zeroize/latest/zeroize/struct.Zeroizing.html

```suggestion
        let mut data_key = Zeroizing::new(key::derive_phase_key(&side, &key, &phase));
```

Reply:
Cool, could I change the signature of `derive_phase_key` to do that automatically for everything it returns? It currently returns a `Vec<u8>`.

Reply:
what would the new signature be? `-> Zeroizing<Vec<u8>>` ?

Reply:
Yup

Reply:
ok, dumb question, what's the cleanest way to convert the unit test at line 280-ish, from `assert_eq!(hex::encode(dk11), "abcabc..")`? That `dk11` was a `Vec<u8>`, now it's a `Zeroizing<Vec<u8>>`. The compiler is complaining that `std::convert::AsRef<[u8]>` is not implemented (`hex::encode` is `pub fn encode<T: AsRef<[u8]>>(data: T) -> String`).

Reply:
huh, `hex::decode(&*dk11)` worked, but surely that's silly?

Reply:
It's doing deref coercion

Reply:
ok, so not silly (or at least not too silly to use). thanks!